src/Contexts/Adaptors/EVMAdaptors/EVMDestinationAdaptorProvider.tsx

```diff
@@ -145,6 +153,60 @@ export const EVMDestinationAdaptorProvider = ({
     tokensDispatch,
   ]);
 
+  const initFallbackMechanism = useCallback(async (): Promise<void> => {
+    setFallbackInitialized(true);
+    const srcChainId = homeChainConfig?.chainId as number;
+    const destinationChainId = destinationChainConfig?.chainId as number;
+    const { delayMs, delayRatio } = getСhainTransferFallbackConfig(
+      srcChainId,
+      destinationChainId
+    );
+    const erc20ProposalHash = getErc20ProposalHash(
+      (destinationChainConfig as EvmBridgeConfig).erc20HandlerAddress,
+      depositAmount as number,
+      depositRecipient as string
+    );
+
+    fallback(delayMs, delayRatio, async () => {
+      const res = await destinationBridge?.getProposal(
+        srcChainId,
+        parseInt(depositNonce as string),
+        erc20ProposalHash
+      );
+      const status = res ? res[4] : undefined;
+      console.log("Proposal votes status", status);
+      switch (status) {
+        case VoteStatus.EXECUTED:
+          setTransactionStatus("Transfer Completed");
+          setFallbackInitialized(false);
+          return false;
+        case VoteStatus.CANCELLED:
+          setTransactionStatus("Transfer Aborted");
+          setFallbackInitialized(false);
+          return false;
+        default:
+          return true;
+      }
+    });
+  }, [
+    homeChainConfig,
+    destinationChainConfig,
+    depositRecipient,
+    depositNonce,
+    depositAmount,
+    destinationBridge,
+  ]);
+
+  useEffect(() => {
+    console.log({ transactionStatus }); // ToDo: check why get transaction status update several times on the same status
+    if (
+      transactionStatus === "In Transit" &&
```

Comment:
We should not depend on `In Transit` here.

Reply:
Resolved